### PR TITLE
feat(portal): expose overflow boundary options on flip and shift

### PR DESCRIPTION
**Pull request by ashley-hunter**
## PR Checklist

- [x] The commit message follows our guidelines: https://github.com/ng-primitives/ng-primitives/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## Issue

Related to #689 — see the note below on why this does not close it outright.

## What does this PR implement/fix?

Split out of #862 so it can be reviewed on its own.

`NgpFlipOptions` and `NgpShiftOptions` gain three options that Floating UI's `flip` and `shift` already support but that the overlay had no way to pass through:

- **`boundary`** — the clipping area the overlay is kept within. Pass an element (or elements) to constrain an overlay to a container rather than to its clipping ancestors.
- **`rootBoundary`** — measure against the document instead of the viewport.
- **`crossAxis`** — widen the axis checks. `boolean | 'alignment'` for flip, matching its three-way semantics; plain `boolean` for shift, where the default is `false`.

`size` now shares whichever boundary is in effect. It reports the space available before the overlay overflows, so measuring it against a different region than the one keeping the overlay in view made `--ngp-*-available-width` / `-height` describe an area the overlay was never constrained to. The boundary and rootBoundary are taken as a pair from flip when flip names one, otherwise from shift — enabled-by-default flip resolves to `{}`, so the test is whether it *names* a boundary rather than whether it exists, or `{}` would mask a boundary set only on shift. Padding is deliberately not shared: adopting it would shrink the reported dimensions for anyone already setting it.

Docs for all three options are added to the menu, popover and tooltip pages, and the existing `api-css-var` descriptions for the available-width/height variables are corrected — they claimed the viewport, which is now only the default.

### Relationship to #689

#689 reports that a portalled overlay whose trigger sits in a scroll container flips against the viewport rather than that container, and proposes defaulting flip's boundary to the trigger's overflow ancestors.

This PR does not change that default. What it does is make the issue's workaround unnecessary: instead of disabling the built-in flip and supplying custom middleware through `updateConfig`, the boundary is now a supported option —

```html
<button [ngpMenuTrigger]="menu" [ngpMenuTriggerFlip]="{ boundary: scrollContainer }">…</button>
```

and `size` measures against that same boundary, which is the other half of the workaround in the issue.

Defaulting the boundary to the trigger's overflow ancestors is a behaviour change for every existing consumer with a scroll container, so it is deliberately left as a separate decision rather than folded into an additive PR.

## Testing

38 tests across `overlay-boundary.test.ts` (new), `flip.test.ts` (new — `coerceFlip` had no coverage) and `shift.test.ts`. Each was verified to fail without its change. Notably the shift-boundary case has two tests: one reaching the branch with `flip: false`, and one leaving flip at its default, which is the case consumers actually hit and which the first cut got wrong.

Full suite green: 2417 browser tests, build and lint clean.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

All three options are optional and default to Floating UI's existing behaviour.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01VWM2JuKvvtBwrVxXwoMhYu

## Summary by CodeRabbit

* **New Features**
  * Added configurable overflow boundaries for menus, popovers and tooltips.
  * Added support for viewport, document, container and scroll-container boundaries.
  * Added controls for cross-axis flipping and shifting, including trigger-based boundary measurement.
  * Available width and height values now reflect the active overflow boundary.

* **Documentation**
  * Expanded guidance and examples covering boundary, root boundary, cross-axis and available-space behaviour.

### CODE REVIEW

File: apps/documentation/src/app/pages/(documentation)/primitives/tooltip.md [hunk 1]
```diff
@@ -106,6 +106,46 @@ You can customize the shift behavior to control how the tooltip stays within the
...
+  </button>
+</div>
+
+<!-- Measure against the whole document rather than the viewport -->
+<button [ngpTooltipTrigger]="tooltip" [ngpTooltipTriggerFlip]="{rootBoundary: 'document'}">
+  Tooltip that does not flip while off-screen
+</button>
+```
+
+`altBoundary` measures against the trigger's clipping ancestors rather than the tooltip's. The tooltip is portalled to the body, so its own clipping ancestors are effectively the viewport - set this when the trigger sits in a scroll container.
```
[the remark this answers is not in the document]
Comment (ashley-hunter) [reply apps/documentation/src/app/pages/(documentation)/primitives/tooltip.md]: Confirmed and fixed in 60d7e59 — good catch.

Verified against `getClippingRect` in the installed `@floating-ui/dom` rather than the docs:

```js
const elementClippingAncestors =
  boundary === 'clippingAncestors'
    ? (isTopLayer(element) ? [] : getClippingElementAncestors(element, this._c))
    : [].concat(boundary);
```

`altBoundary` chooses which element `element` is, but that only feeds the clipping-ancestor walk on the `'clippingAncestors'` branch. With an explicit element, array or rect, `[].concat(boundary)` is used verbatim and the choice of element no longer affects the region. So `{boundary, altBoundary: true}` measures against `boundary`, as you say.

The paragraph now ends with "It applies only while `boundary` is left at its default; an explicit boundary always wins." on all three pages, and the same clause is on the `altBoundary` JSDoc in `flip.ts` and `shift.ts`.

---

On the second finding, about `--ngp-tooltip-available-width` / `-height` at lines 239-240 — skipping, as it is already done. Those descriptions read:

> The available width of the tooltip before it overflows the flip boundary, or the shift boundary when flip does not set one. Defaults to the viewport.

which already states the selected clipping context and the viewport default; they were corrected earlier in this PR. (`flippedBoundary` in the suggestion is not an option this library or Floating UI exposes.)

---
_Generated by [Claude Code](https://claude.ai/code)_
